snooty/diagnostics.py

```diff
@@ -46,6 +48,14 @@ def severity(self) -> "Diagnostic.Level":
     def severity_string(self) -> str:
         return self.severity.name.title()
 
+    def serialize(self) -> n.SerializedNode:
+        diag: Dict[str, SerializableType] = {}
+        """Create dict containing diagnostic attributes for neatly reporting diagnostics at program completion"""
```

Comment:
Docstrings go right after the function header; that's what tells Python to attach it to the function object